spacepy/pybats/__init__.py

```diff
@@ -1742,21 +1771,61 @@ def __init__(self, filename=False, load=True, npoints=0, *args, **kwargs):
         if filename and load:  # Load contents from existing file.
             self.read(filename)
         else:
-            units   = ['nT', 'nT', 'nT', 'km/s', 'km/s', 'km/s', 'cm^-3', 'K']
+            units = ['nT', 'nT', 'nT', 'km/s', 'km/s', 'km/s', 'cm^-3', 'K']
             for i, key in enumerate(self.attrs['var']):
-                self[key]=dmarray(zeros(npoints), attrs={'units':units[i]})
+                self[key] = dmarray(zeros(npoints), attrs={'units': units[i]})
 
         # Determine the density variable, which can either be "n" or "rho".
```

Comment:
Is this ever the case? Is this done in a subclass or something? It looks like things are consistently rho (I'm just going to quietly scream in the corner about using rho for a number density) and the docs need to be updated.

Reply:
Yes- if you create a "blank" object (one where you don't load from file immediately). You would do this when, for example, building an ImfInput object by filling in the data from Omni or some other source.

Reply:
If you make a blank object then the density variable is n? Where does this happen? It looks like it's made from looping over self.attrs['var'] which has rho but not n.

Reply:
rho vs n has been fixed (per comment above), so it will be n on a black object. `self.attrs['var']` issues has been updated, no more rho.
Docs updated.

Reply:
Yes, if you make a blank object the density variable is 'n'. `self.attrs['var']` has been updated per the above comment on variable naming.